backend/app/controllers/spree/admin/promotion_code_batches_controller.rb

```diff
@@ -0,0 +1,31 @@
+module Spree
+  module Admin
+    class PromotionCodeBatchesController < ResourceController
+      belongs_to 'spree/promotion'
+
+      create.after :build_promotion_code_batch
+
+      def download
+        require "csv"
+
+        @promotion_code_batch = Spree::PromotionCodeBatch.find(
+          id: params[:promotion_code_batch_id]
+        )
+
+        respond_to do |format|
+          format.csv do
+            filename = "promotion-code-batch-list-#{@promotion_code_batch.id}.csv"
+            headers["Content-Type"] = "text/csv"
+            headers["Content-disposition"] = "attachment; filename=\"#{filename}\""
```

Comment:
Why not use [Rails' `send_file`](rails doc send_file) here? Then you can spare to set headers, Rails does this for you.